src/Controller/OmniaController.php

```diff
@@ -141,282 +268,218 @@ public function videoList() {
    * @throws \Exception
    */
   public function video(Request $request) {
-
-    $response = new JsonResponse();
-    $content = $request->getContent();
-    $query = $this->mediaEntityStorage()->getQuery();
-
-    $token = $request->query->get('token', NULL);
-    if (NULL !== $token) {
-
-      $config = $this->config('nexx_integration.settings');
-      if ($token != $config->get('notification_access_key')) {
-        throw new AccessDeniedHttpException();
-      }
-    }
-    else {
+    if (!$this->isTokenValid($request)) {
+      $this->logger->error('Wrong or missing token in request.');
       throw new AccessDeniedHttpException();
     }
 
-    if (!empty($content)) {
-      $videoData = json_decode($content);
-    }
-    if (!isset($videoData->itemID)) {
+    $content = $request->getContent();
+    $this->videoData = json_decode($content);
```

Comment:
Add a check whether the decode was successful. If not, throw HttpException with status code 503. Guess in this case a log entry would make sense then, because this should be exceptional - e.g. when Nexx API has changed again.